Project Finished

CarDioLogic opened the pull request.

CarDioLogic committed:
Project Finished

TheCSharpAcademy reviewed (changes requested):
@CarDioLogic Thanks for submitting!

I've had this issue when filtering: 

![image](https://github.com/TheCSharpAcademy/CodeReviews.MVC.Budget/assets/108711476/03f870a6-bc37-47c1-829b-e5e148f76b0a)

And a couple of minor issues in the ui:
Filterid label: 
![image](https://github.com/TheCSharpAcademy/CodeReviews.MVC.Budget/assets/108711476/73bae7b4-35b2-4983-9cae-a643d2bea821)

Release date label:
![image](https://github.com/TheCSharpAcademy/CodeReviews.MVC.Budget/assets/108711476/c471d29d-9abb-4579-93fe-306e6d193ab8)

CarDioLogic committed:
Corrected the filter function to filter properly and made it return the correct model to the view.
Changed the "release date" in the UI to "Transaction Date".
Removed some unnecesary code (e.g an Id property from the transactionViewModel), and adapted the Delete method in the transactions controller to it.
